mcx/diagnostics/gelman_rubin.py

```diff
@@ -1,18 +1,113 @@
 """Kernel to compute the Gelman-Rubin convergence diagnostic (Rhat) online.
 """
-from typing import NamedTuple
+from typing import Callable, NamedTuple, Tuple
 
+import jax
 import jax.numpy as jnp
 
-from mcx.inference.warmup.mass_matrix_adaptation import (
-    WelfordAlgorithmState,
-    welford_algorithm,
-)
+
+class WelfordAlgorithmState(NamedTuple):
+    """State carried through the Welford algorithm.
+
+    mean
+        The running sample mean.
+    m2
+        The running value of the sum of difference of squares. See documentation
+        of the `welford_algorithm` function for an explanation.
+    sample_size
+        The number of successive states the previous values have been computed on;
+        also the current number of iterations of the algorithm.
+    """
+
+    mean: float
+    m2: float
+    sample_size: int
 
 
 class GelmanRubinState(NamedTuple):
     w_state: WelfordAlgorithmState
-    rhat: float
+    rhat: jnp.DeviceArray
+    worst_rhat: jnp.DeviceArray
```

Comment:
I think the worst rhat should be computed in the inference loop since it is specific to this use, and we don't need it _in general_.

Reply:
Is that me or you are computing it in the loop but did not remove the computation inside `update` and the tuple element in `GelmanRubinState` ?